### ISSUE TITLE
Why need multiple calls to filter_rects() with different thresholds?

### ISSUE THREAD

**Issue by atique81** [Russell91/TensorBox#149]
Hi @Russell91
Could you please explain the intuition behind the multiple calls to the method filter_rects() with different thresholds inside the stitch_wrapper.pyx (https://github.com/Russell91/TensorBox/blob/master/utils/stitch_wrapper.pyx#L73)?
Your paper does not seem to have any explanation, or even mention of this!

**Comment by Russell91** [comment]
Yea, in ordinary bounding box merging, we always accept boxes with the highest threshold first. The multi-stage merging process approximates this in spite of the fact that more complicated thingsn are going on.

The specific problem you are trying to avoid is two regions having boxes

A1: 0.9  A2: 0.1
B1: 0.1   B2: 0.9

Where A1 and B1 are merged, and A2 and B2 are merged, in spite of the fact that A1 and B2 are actually the same boxes. The reason why this could be an issue is that the a single iteration of filter_rects ignores the confidences (and this is actually a somewhat good idea as the statistics of the confidences can vary a lot between problems). So since you ignore confidences in a single iteration, you can get the problem above, where you'd like to end up with one 0.9 confidence box, and one 0.1 confidence box, but instead you end up with two 0.9 confidence boxes (and you get a false positive early on the PR curve!).

Hope that begins to make sense.

**Comment by atique81** [comment]
Thanks @Russell91 for the explanation, though it seems to be complicated to me. I do not find any mention of this process in your paper, is there?

**Comment by Russell91** [comment]
Sometimes for complicated processes like these, it's better to describe the high level in the paper and then just release the code. The explanation I gave above is actually the most detailed written version to date. If there are further questions, please feel free to ask them though.

**Comment by atique81** [comment]
Yeah, I agree with you. Thanks for the nice explanation of the process above!
